my-app/src/Utilities/ValidationUtilities.tsx

```diff
@@ -123,6 +125,42 @@ export function isValidMove(
                .map(point => getPieceAtLocation(point))
                   .some(piece => piece !== null);      
    }
+
+   /*  
+      castling is only possible if neither the king nor the rook has moved
+      there must not be any pieces between the king and the rook
+      the king may not be in check
+      the square the king goes to and any intervening squares may not be under attack
+      */
+   function pieceIsCastlingAndCannotCastle(moveLocation: Coordinate, piece: Piece, board: Array<Array<(Piece | null)>>) : boolean {
```

Comment:
Is isValidCastlingMove an ok name for this function?

Reply:
I guess we will also need to reuse this function in the handleMove function to move both the king and the castle on the board.

Reply:
works for me. Yeah I think we will also need to add a check in Chess.tsx that uses this and moves both pieces if its castling